src/core/device.c

```diff
@@ -859,9 +865,18 @@ static void device_enumerate(Manager *m) {
                 goto fail;
         }
 
+        /* At this point we don't really know if udev is running and has populated its DB
+         * because reload of PID1 can happen at anytime especially early during the boot
+         * process. So in order to figure this out, we simply assume that if the DB is
+         * fully populated, we should at least enumerate one device otherwise we consider
+         * that the device enumeration hasn't happened yet hence we shouldn't trust it */
```

Comment:
Is the assumption true even if there exist a rule which has `db_persistent`?

Reply:
One more question, what happens when udev is processing devices, that is, some devices are initialized, but some are not.